src/core/execute.c

```diff
@@ -4262,10 +4262,10 @@ void exec_context_dump(const ExecContext *c, FILE* f, const char *prefix) {
         if (exec_context_restrict_namespaces_set(c)) {
                 _cleanup_free_ char *s = NULL;
 
-                r = namespace_flag_to_string_many(c->restrict_namespaces, &s);
+                r = namespace_flags_to_string_alloc(c->restrict_namespaces, &s);
```

Comment:
Is the "_alloc" part actually needed? Usually functions which return a static string do it through the return value, and functions which construct the answer, do it through an output argument, so "_alloc" is not needed for the reader to know that the answer needs to be freed.